### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by gfronza**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,10 +70,7 @@ calculatorService.Register("subtract", func(req server.Request, res *server.Resp
     // subtraction logic here...
 })
 
-calculatorService.Start()
```
Comment (gfronza) [inline README.md:73]: @alexandrevicenzi this method was replaced by `ListenAndServe` that is sync. You'll need to use `go service.ListenAndServe()` to continue the execution.

**Comment by gfronza** [comment]
@skrater @alexandrevicenzi

File: README.md (hunk 1 above)
Comment (alexandrevicenzi) [reply README.md:73]: That's a shame :/ It will break all services.

File: _examples/server/example_server.go [hunk 2]
```diff
@@ -11,16 +11,16 @@ import (
 
 func main() {
 	b, err := broker.NewBroker(os.Getenv("AMQP_URL"))
+	defer b.Close()
```
Comment (alexandrevicenzi) [inline _examples/server/example_server.go]: so, if we got an error b will not be nil?

Comment (alexandrevicenzi) [reply _examples/server/example_server.go]: looking into the code this can panic

File: broker/broker.go [hunk 3]
```diff
@@ -1,44 +1,126 @@
...
+// NotifyReestablish returns a channel to notify when the connection is restablished.
+func (b *Broker) NotifyReestablish() <-chan bool {
+	receiver := make(chan bool)
+	b.reestablishs = append(b.reestablishs, receiver)
+
+	return receiver
+}
+
 // WaitUntilConnectionCloses hold the current goroutine until the connection with the broker closes.
 func (b *Broker) WaitUntilConnectionCloses() {
```
Comment (alexandrevicenzi) [inline broker/broker.go]: why do we have this? it's not used anymore, ritght?

File: server/metrics_test.go [hunk 4]
```diff
@@ -35,7 +35,7 @@ func TestMetricsShipperExtension(t *testing.T) {
 	shippedMetricsCount := 0
 
 	go func() {
-		for _ = range dc {
+		for range dc {
```
Comment (alexandrevicenzi) [inline server/metrics_test.go:36]: very nice indeed

**Review by alexandrevicenzi — APPROVED** [review]
The broker has no NotifyClose.

The OpenChannel is public, that's not good.

**Review by alexandrevicenzi — CHANGES_REQUESTED** [review]
Fix the defer.

**Comment by gfronza** [comment]
Fixes https://github.com/porthos-rpc/porthos-go/issues/45

**Pull request merged**